main/lib/src/LogSender.cc

```diff
@@ -10,6 +10,12 @@ namespace eudaq {
     m_logclient(0), m_errlevel(Status::LVL_DEBUG), m_shownotconnected(false) {}
 
   void LogSender::Connect(const std::string & type, const std::string & name, const std::string & server) {
+    MutexLock m(m_mutex);
+    if (isConnected)
+    {
```

Comment:
Minor style issue:
in other code around we don't have dangling parentheses, maybe also write
```
if() {
```
instead of
```
if()
{
```